telegram/ext/_handlers/callbackqueryhandler.py

```diff
@@ -141,9 +166,12 @@ def check_update(self, update: object) -> Optional[Union[bool, object]]:
         # pylint: disable=too-many-return-statements
         if isinstance(update, Update) and update.callback_query:
             callback_data = update.callback_query.data
-            if self.pattern:
-                if callback_data is None:
-                    return False
+            game_short_name = update.callback_query.game_short_name
+
+            if not any([self.pattern, self.game_pattern]):
+                return True
+
+            if callback_data:
```

Comment:
I guess this'll need to stay a "if self pattern" along with returning False if callback_data is None. Currently, you'll run into problems if self.pattern is None in line 181

Same hold for the game-clause below

Reply:
https://github.com/python-telegram-bot/python-telegram-bot/pull/4353/commits/5cac77868078edc01d7c9ad701dec0d46655cd83 should solve that problem.

Reply:
yes, adding `if not self.pattern` solves the problem, but tbh I find it rather confusing to read. IMO the if-else logic sohuld follow the logic of "what do I cant to allow?" (i.e. (game_)pattern) rather than "what's available?" (i.e. data/game_short_name). Note that you already start in that direction with `if not any([self.pattern, self.game_pattern]):`. After that check it seems more logical to me to have `if self.pattern: … elif self.game_pattern` as the next logical step.

Reply:
How can I implement XOR-like behavior based on an incoming update? If the user provides both `pattern` and `game_pattern`, CQH will always prioritise the first condition. 

To overcome this case, I used if-else to check "what's available". 

Let me know if this is not needed. I can change it to ```if self.pattern: … elif self.game_pattern```

Reply:
You're right. Thanks for getting me on track!

the if-then-else causes could use some CSI-style comments, then as linked in the contribution guide.

btw, if you could change line 154 to 

```python
if not (isinstance(update, Update) and update.callback_query):
    return False
```

that can save us some indentation & nesting…

Reply:
I returned `None` in line 154 because thats what the code was doing before and imo it doesn't change anything as it will still stay at "don't handle" 

Also left comments, let me know if they look good.